### PR TITLE
White label fixes

### PR DESCRIPTION
**Pull request by paullinator**
### CHANGELOG

 none

### Dependencies

https://github.com/EdgeApp/edge-login-ui-rn/pull/80

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [x] Tested on iOS device
- [ ] Tested on Android device
- [x] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204732626349711

### CODE REVIEW

File: src/locales/en_US.ts [hunk 1]
```diff
@@ -109,10 +109,10 @@ const strings = {
   warning_please_wait_title: 'Please Wait',
 
   warning_scam_title: 'Scam Warning',
-  warning_scam_message_financial_advice: 'Edge will not give financial advice.',
+  warning_scam_message_financial_advice_s: '%s will not give financial advice.',
   warning_scam_message_irreversibility: 'Cryptocurrency transactions are irreversible.',
   warning_scam_message_unknown_recipients: 'Do not send money to people or organizations you do not know.',
-  warning_scam_footer: 'If you have any questions or concerns regarding this send please contact support@edge.app',
+  warning_scam_footer_s: 'If you have any questions or concerns regarding this send please contact %s',
```
Comment (samholmes) [inline src/locales/en_US.ts:115]: `If you have any questions or concerns regarding this send, please contact %s`

File: src/components/scenes/LoginScene.tsx [hunk 2]
```diff
@@ -152,9 +152,11 @@ class LoginSceneComponent extends React.PureComponent<Props, State> {
...
   }
 
-  onComplete = () => {
-    this.props.navigation.navigate('gettingStarted', {})
-  }
+  onComplete = ENV.USE_WELCOME_SCREENS
+    ? () => {
+        this.props.navigation.navigate('gettingStarted', {})
+      }
+    : undefined
```
Comment (samholmes) [inline src/components/scenes/LoginScene.tsx:159]: It's purely preference, but I would have put the ternary at the JSX attribute instead of the definition of the method itself.

File: src/__tests__/components/TransactionListTop.test.tsx [hunk 3]
```diff
@@ -52,7 +52,8 @@ describe('TransactionListTop', () => {
     core: {
       account: {
         currencyWallets: { '123': fakeWallet },
-        currencyConfig: { bitcoin: fakeCurrencyConfig }
+        currencyConfig: { bitcoin: fakeCurrencyConfig },
+        watch() {}
```
Comment (samholmes) [inline src/__tests__/components/TransactionListTop.test.tsx:56]: Why is this `watch()` added for this commit? Seems unrelated; not sure what's going on here.

File: src/components/themed/SideMenu.tsx [hunk 4]
```diff
@@ -266,7 +266,7 @@ export function SideMenu(props: DrawerContentComponentProps) {
     }
   ]
 
-  if (IONIA_SUPPORTED_FIATS.includes(defaultFiat)) {
+  if (ENV.ENABLE_VISA_PROGRAM && IONIA_SUPPORTED_FIATS.includes(defaultFiat)) {
```
Comment (samholmes) [inline src/components/themed/SideMenu.tsx:269]: We'll need to add this flag to the `TransactionListTop` scene too.

File: src/util/CurrencyInfoHelpers.ts [hunk 5]
```diff
@@ -2,12 +2,20 @@ import { EdgeAccount, EdgeCurrencyInfo, EdgeCurrencyWallet, EdgeToken } from 'ed
...
   .map(pluginId => SPECIAL_CURRENCY_INFO[pluginId].chainCode)
-const keysOnlyModePlugins = Object.keys(SPECIAL_CURRENCY_INFO).filter(pluginId => SPECIAL_CURRENCY_INFO[pluginId].keysOnlyMode ?? false)
+const possibleKeysOnlyPlugins = [...new Set([...Object.keys(SPECIAL_CURRENCY_INFO), ...Object.keys(ENV.KEYS_ONLY_PLUGINS)])]
+export const keysOnlyModePlugins = possibleKeysOnlyPlugins.filter(pluginId => {
+  if (ENV.KEYS_ONLY_PLUGINS[pluginId]) {
+    return true
+  } else {
+    return SPECIAL_CURRENCY_INFO[pluginId]?.keysOnlyMode ?? false
+  }
+})
```
Comment (samholmes) [inline src/util/CurrencyInfoHelpers.ts:18]: Why hoge-poge with a deprecated flag; why not go full send and remove the field from SPECIAL_CURRENCY_INFO?

**Review by samholmes — CHANGES_REQUESTED** [review]
Some little question/comments/feedback.

File: src/components/scenes/LoginScene.tsx (hunk 2 above)
Comment (paullinator) [reply src/components/scenes/LoginScene.tsx:159]: maybeHandleComplete

File: src/__tests__/components/TransactionListTop.test.tsx (hunk 3 above)
Comment (paullinator) [reply src/__tests__/components/TransactionListTop.test.tsx:56]: context

**Review by samholmes — APPROVED** [review]

File: scripts/deploy.ts [hunk 6]
```diff
@@ -164,6 +164,8 @@ function buildIos(buildObj: BuildObj) {
     process.env.MATCH_PASSWORD != null
   ) {
     const githubSshKey = process.env.GITHUB_SSH_KEY ?? join(_rootProjectDir, 'id_github')
+    call(`security unlock-keychain -p '${process.env.KEYCHAIN_PASSWORD || ''}' "${process.env.HOME || ''}/Library/Keychains/login.keychain"`)
+    call(`security set-keychain-settings -l ${process.env.HOME || ''}/Library/Keychains/login.keychain`)
```
[the remark this answers is not in the document]
Comment (paullinator) [reply scripts/deploy.ts]: fixed

File: src/locales/en_US.ts (hunk 1 above)
Comment (paullinator) [reply src/locales/en_US.ts]: fixed

File: src/__tests__/components/TransactionListTop.test.tsx (hunk 3 above)
Comment (paullinator) [reply src/__tests__/components/TransactionListTop.test.tsx:59]: fixed
